packages/playwright/src/reporters/html.ts

```diff
@@ -418,8 +417,7 @@ class HtmlBuilder {
         projectName,
         location,
         duration,
-        // Annotations can be pushed directly, with a wrong type.
-        annotations: test.annotations.map(a => ({ type: a.type, description: a.description ? String(a.description) : a.description })),
+        annotations: this._serializeAnnotations(test.annotations),
```

Comment:
Does `annot:foo` still filter by dynamic annotations? Since we are not adding them here anymore, I think not.